src/Diagnostics.CompilerHost/Startup.cs

```diff
@@ -21,9 +29,14 @@ public class Startup
         /// Initializes a new instance of the <see cref="Startup"/> class.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
-        public Startup(IConfiguration configuration)
+        public Startup(IHostingEnvironment hostingEnvironment)
         {
-            Configuration = configuration;
+            var builder = new ConfigurationBuilder()
+              .SetBasePath(Directory.GetCurrentDirectory())
+              .AddJsonFile($"appsettings.json", optional: false, reloadOnChange: true)
```

Comment:
I believe I read this somewhere but doing `.AddJsonFile` followed by `.AddEnvironmentVariables` means the .NET Core runtime will first go to appsettings.json file to get settings and if not found then environment variables next. I was working on a project onetime and I deployed my .NET Core app with Visual Studio publish which included the appsettings.json file. When I ran the web app, I was confused where it was getting these different settings than what I had set in the App Service AppSettings. Its not the biggest deal but it might save a headache in the future by changing the order.